sshd-core/src/main/java/org/apache/sshd/common/channel/ChannelOutputStream.java

```diff
@@ -50,12 +64,15 @@ public class ChannelOutputStream extends OutputStream implements java.nio.channe
     private final Duration maxWaitTimeout;
     private final byte cmd;
     private final boolean eofOnClose;
-    private final byte[] b = new byte[1];
-    private final AtomicBoolean closedState = new AtomicBoolean(false);
+    private final AtomicReference<OpenState> openState = new AtomicReference<>(OpenState.OPEN);
+
+    private final Object bufferLock = new Object();
     private Buffer buffer;
     private int bufferLength;
     private int lastSize;
-    private boolean noDelay;
+    private boolean isFlushing;
+
+    private volatile boolean noDelay;
```

Comment:
Personally I am not a big fan of *volatile* as means  of multi-threading safety - I prefer *AtomicBoolean*-s - but I defer to your judgement here if you prefer *volatile*

Reply:
Done.